lib/tpm2.c

```diff
@@ -1065,6 +1065,12 @@ uint32_t fix_esys_hierarchy(TPMI_RH_HIERARCHY hierarchy)
 {
 #if defined(ESYS_3_0)
     switch (hierarchy) {
+        case ESYS_TR_RH_NULL:
+        case ESYS_TR_RH_OWNER:
+        case ESYS_TR_RH_ENDORSEMENT:
+        case ESYS_TR_RH_PLATFORM:
+        case ESYS_TR_RH_PLATFORM_NV:
```

Comment:
You need falls-thru comments so the compiler won't complain, CFLAGS `-Wextra` enables `-Wimplicit-fallthrough=3`. Details can be found here:
  - https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html

Some versions of gcc/clang are more pedantic than others, but I know occasionally Fedora folks will come back with patches. This may be better for an if/elseif type of thing now.

Reply:
I thought `/* FALLTHRU */` was only needed where the `case` statements were not in consecutive lines ?

Reply:
Oh yeah, I didn't realize that. It's only really if theirs some code in the statements....